test/core/iomgr/resolve_address_posix_test.cc

```diff
@@ -59,6 +61,9 @@ void args_init(args_struct* args) {
 
 void args_finish(args_struct* args) {
   GPR_ASSERT(gpr_event_wait(&args->ev, test_deadline()));
+  args->thd.Join();
+  // Don't need to explicitly destruct args->thd since
+  // args is actually going to be destructed, not just freed
```

Comment:
Where are `args` going to be destructed ? (it might help to cross reference the code here) - or call `~Thread()` anyway once since it is safe to call `~Thread()` multiple times.

Reply:
Note that at each of the call-sites, `args_finish` is called with a stack-allocated `args` structure, not a struct that is allocated on the heap with `gpr_malloc`.

Reply:
I've also removed the use of explicit destruction now